### PR TITLE
feat: add OpenTelemetry tracing

### PR DESCRIPTION
**Pull request by olavloite**
Adds `OpenTelemetry` tracing to the JDBC driver. The `OpenTelemetry` instance that should be used for tracing should be either be set as the global `OpenTelemetry` instance, or set in the `Properties` map that is used to create a JDBC connection. In the latter case, the specific `OpenTelemetry` instance will only be used for that connection.

### CODE REVIEW

File: .readme-partials.yaml [hunk 1]
```diff
@@ -46,6 +48,106 @@ custom_content: |
...
+                            .put("service.name", "my-unique-service-name")
+                            .build())
+                    .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                    .build())
+            .buildAndRegisterGlobal();
+   String projectId = "my-project";
+   String instanceId = "my-instance";
+   String databaseId = "my-database";
+   // Setting this to true instructs the JDBC driver to include the SQL statement with the traces.
+   boolean enableExtendedTracing = true;
```
Comment (surbhigarg92) [inline .readme-partials.yaml:87]: Should we also `enableApiTracing` in the code snipper

File: .readme-partials.yaml [hunk 2]
```diff
@@ -46,6 +48,106 @@ custom_content: |
...
+           System.out.printf(
+               "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+         }
+       }
+     }
+   }
+   ```
+   
+   #### Example Using an OpenTelemetry instance in Properties
+   Create and register a global OpenTelemetry object before creating a JDBC connection.
```
Comment (surbhigarg92) [inline .readme-partials.yaml:106]: We are not registering global OT object here. 

nit: I feel we can simplify this example by not repeating the whole code . 

We can create OpenTelemetrySdkBuilder object and then show different usage of it. 

for eg. 

```
 OpenTelemetrySdkBuilder openTelemetrySdkBuilder =
        OpenTelemetrySdk.builder()
            .setTracerProvider(
                SdkTracerProvider.builder()
                    .setSampler(Sampler.traceIdRatioBased(0.05))
                    .setResource(
                        Resource.builder()
                            .put("service.name", "my-unique-service-name")
                            .build())
                    .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
                    .build());
 // Example Using Global OpenTelemetry
 openTelemetrySdkBuilder.buildAndRegisterGlobal()
// Example Using an OpenTelemetry instance in Properties 
     OpenTelemetry openTelemetry = openTelemetrySdkBuilder.build()
  ```

File: documentation/latency-debugging-guide.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
+# Spanner JDBC Driver - Latency Debugging Guide
+
+The Spanner JDBC driver supports OpenTelemetry tracing. Tracing can be used to investigate slow
```
Comment (surbhigarg92) [inline documentation/latency-debugging-guide.md:3]: We also have metrics which is helpful in debugging latency issues. Should that info also be added here https://cloud.google.com/spanner/docs/latency-points

File: documentation/latency-debugging-guide.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+       System.out.printf(
+           "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+     }
+   }
+ }
+}
+```
+
+### Example Using an OpenTelemetry instance in Properties
+Create and register a global OpenTelemetry object before creating a JDBC connection.
```
Comment (surbhigarg92) [inline documentation/latency-debugging-guide.md:83]: Here as well. global OT is not used here

File: documentation/latency-debugging-guide.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+     while (rs.next()) {
+       System.out.printf(
+           "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+     }
+   }
+ }
+}
+```
+
+## Traces
```
Comment (surbhigarg92) [inline documentation/latency-debugging-guide.md:128]: This only talks about JDBC traces, but in reality customer would see way more traces coming from spanner library.

File: .readme-partials.yaml (hunk 1 above)
Comment (olavloite) [reply .readme-partials.yaml:87]: Added (although we need to wait for https://github.com/googleapis/java-spanner/pull/3166 to be merged before we can use it here)

File: .readme-partials.yaml (hunk 2 above)
Comment (olavloite) [reply .readme-partials.yaml:106]: Thanks for noting. Removed the text regarding registering a global OTEL object.

I also simplified the second example a bit, assuming that readers will understand that they can use the same OpenTelemetry instance as in the example above.

File: documentation/latency-debugging-guide.md (hunk 3 above)
Comment (olavloite) [reply documentation/latency-debugging-guide.md:3]: Added a link to the page. The intention is to also add this guide to the documentation (probably to the same submenu)

File: documentation/latency-debugging-guide.md (hunk 4 above)
Comment (olavloite) [reply documentation/latency-debugging-guide.md:83]: Removed

File: documentation/latency-debugging-guide.md (hunk 5 above)
Comment (olavloite) [reply documentation/latency-debugging-guide.md:128]: This text 'pretends' that all the traces come from the JDBC driver (although some of the text also mentions the client lib). I think that for a customer it is easier to read/understand if we mainly talk about JDBC here, as most users who use the JDBC driver are not using the client library. The fact that they are layered is not necessarily a known fact for everyone.
The traces that are described here also include the traces from the client library (e.g. everything in point 2 and 3 are actually collected by the client library)

**Review by surbhigarg92 — APPROVED** [review]

File: documentation/latency-debugging-guide.md (hunk 5 above)
Comment (olavloite) [reply documentation/latency-debugging-guide.md:128]: Actually, those are the ones that are described in point 2 below, but the text did not call out what they are named. I added that to the text to make it easier for a reader to make the link.

**Pull request merged**